xwiki-platform-core/xwiki-platform-appwithinminutes/xwiki-platform-appwithinminutes-test/xwiki-platform-appwithinminutes-test-docker/src/test/it/org/xwiki/appwithinminutes/test/ui/AddEntryIT.java

```diff
@@ -17,77 +17,76 @@
+@UITest
+class AddEntryIT
 {
-    @Rule
-    public AdminAuthenticationRule adminAuthenticationRule = new AdminAuthenticationRule(getUtil());
+    private static final String DATE_COLUMN_TITLE = "panel.livetable.doc.date";
 
     /**
-     * The page being tested.
+     * Tests that entry name is URL encoded.
      */
-    private ApplicationHomePage homePage;
-
-    @Before
-    public void setUp() throws Exception
+    @Test
+    @Order(1)
+    void entryNameWithURLSpecialCharacters(TestReference testReference, TestUtils testUtils)
     {
-        getUtil().rest().deletePage(getTestClassName(), getTestMethodName());
-        Map<String, String> editQueryStringParameters = new HashMap<String, String>();
+        // Fixture
+        testUtils.loginAsSuperAdmin();
+
+        testUtils.deletePage(testReference);
+
+        Map<String, String> editQueryStringParameters = new HashMap<>();
         editQueryStringParameters.put("editor", "inline");
         editQueryStringParameters.put("template", "AppWithinMinutes.LiveTableTemplate");
         editQueryStringParameters.put("AppWithinMinutes.LiveTableClass_0_class", "Panels.PanelClass");
-        getUtil().gotoPage(getTestClassName(), getTestMethodName(), "edit", editQueryStringParameters);
+        testUtils.gotoPage(testReference, "edit", editQueryStringParameters);
         // Wait for the page to load before clicking on the save button to make sure the page layout is stable.
-        homePage = new ApplicationHomeEditPage().waitUntilPageIsLoaded().clickSaveAndView();
-    }
+        // The page being tested.
+        ApplicationHomePage homePage = new ApplicationHomeEditPage().waitUntilPageIsLoaded().clickSaveAndView();
 
-    /**
-     * Tests that entry name is URL encoded.
-     */
-    @Test
-    @IgnoreBrowsers({
-    @IgnoreBrowser(value = "internet.*", version = "8\\.*", reason="See https://jira.xwiki.org/browse/XE-1146"),
-    @IgnoreBrowser(value = "internet.*", version = "9\\.*", reason="See https://jira.xwiki.org/browse/XE-1177")
-    })
-    public void testEntryNameWithURLSpecialCharacters()
-    {
+        // Test
         EntryNamePane entryNamePane = homePage.clickAddNewEntry();
         String entryName = "A?b=c&d#" + RandomStringUtils.randomAlphanumeric(3);
         entryNamePane.setName(entryName);
         EntryEditPage entryEditPage = entryNamePane.clickAdd();
         entryEditPage.setValue("description", "This is a test panel.");
         entryEditPage.clickSaveAndView();
 
-        getUtil().gotoPage(getTestClassName(), getTestMethodName());
+        testUtils.gotoPage(testReference);
         homePage = new ApplicationHomePage();
         LiveTableElement entriesLiveTable = homePage.getEntriesLiveTable();
         entriesLiveTable.waitUntilReady();
+        // Makes sure the test entry is displayed at the top of the first page by sorting the date twice (descending
+        // order).
+        entriesLiveTable.sortBy(DATE_COLUMN_TITLE);
```

Comment:
It feels quite wrong to call this twice to get descending order. We'd need a dedicated method with a proper boolean argument to decide if we want sorting ascending or descending. Under the hood it will click twice but we shouldn't see it here.

Reply:
I have introduced two methods to sort a column in ascending and descending order, they are now taking into account the current sorting state of the column and will do between 0 and 2 clicks on the header to set the column in the requested sorting order.
See https://github.com/xwiki/xwiki-platform/pull/1522/files#diff-4d24fd749c8284cd94f3ab972b07f79348fefc0a96d640abf3b12bafde2e025eR323